### ISSUE TITLE
Go: Getting started should set the propagator

### ISSUE THREAD

**Issue by pellared** [open-telemetry/opentelemetry.io#3404]
**What needs to be changed?** 

The getting started examples should enable propagation. While it is not needed for the sake of the example, I got direct feedback from people adopting OpenTelemetry Go that they spent a lot of time finding out that they have to set the propagator.

Reference: https://opentelemetry.io/docs/instrumentation/go/getting-started/#initialize-the-opentelemetry-sdk
 
@open-telemetry/docs-maintainers It would be worth to check if the same problem does not exist for in getting started for other languages.

**Comment by cartermp** [comment]
AFAIK this is mostly a Go problem, but I haven't gone and audited all 11 languages, heh.

**Comment by svrnm** [comment]
It might be an issue for Rust, C++ as well? Not sure.

As an additional note: Propagation and by that correlation of traces across services is a key feature of tracing vs just writing logs, so we should indeed have more specific content around it. For some languages it is hidden in the Manual Instrumentation chapter. JS has a "Context" and "Propagation" page, but both are already very advanced. My thinking was to have a "Context Propagation" page per language that starts with expanding on the "Getting Started" example:

1. Add a new (preinstrumented) service in your particular language that's called "game-service" or something
2. That `game-service` is just calling the `dice-service` twice and then looking who is the winner
3. Look at the combined spans (in console or jaeger?)
4. Add a link to that in the "What's Next" of the Getting Started
5. Put everything else about context & propagation below that

(See https://github.com/pavolloffay/kubecon-eu-2023-opentelemetry-kubernetes-tutorial/blob/main/03-app-instrumentation.md for an example)

wdyt?

**Comment by svrnm** [comment]
> I propose to just ensure that the propagator is set for sake of having a simple getting started.

I agree with that, so having that set properly in the getting started is set for me (and we need eventually to validate other languages as well where we are unsure if this needs to be done)

> The alternative could be to extend the example with simple CLI app which calls the rolldice service using an instrumented HTTP Client instead of using the browser.

That would work as well of course!

**Comment by pellared** [comment]
For now I just propose https://github.com/open-telemetry/opentelemetry-go/pull/4644

In future, I can also try adding a CLI app  which calls the rolldice service using an instrumented HTTP Client instead of using the browser. However, I suggest tracking it as a separate issue as probably this should be something language agnostic as https://github.com/open-telemetry/opentelemetry.io/issues/2623
